src/Xamarin.Android.Build.Tasks/Tasks/BuildApk.cs

```diff
@@ -300,7 +303,11 @@ private void AddAssemblies (ZipArchiveEx apk)
 			IDictionary<string, CompressedAssemblyInfo> compressedAssembliesInfo = null;
 
 			if (compress) {
-				compressedAssembliesInfo = BuildEngine4.GetRegisteredTaskObject (CompressedAssemblyInfo.CompressedAssembliesInfoKey, RegisteredTaskObjectLifetime.Build) as IDictionary<string, CompressedAssemblyInfo>;
+				string key = $"{CompressedAssemblyInfo.CompressedAssembliesInfoKey}:{ProjectFullPath}";
```

Comment:
Given that this string is repeated in two places, could we move it to a method e.g. `MonoAndroidHelper.GetCompressedAssemblyInfoKey(ProjectFullPath)`?

Reply:
Or a `static` method on `CompressedAssemblyInfo` such as `CompressedAssemblyInfo.GetKey (ProjectFullPath)`. Then `CompressedAssembliesInfoKey` could also be `private`.